### PR TITLE
[autoupdate] Add the changelog link for latest_github_release

### PR DESCRIPTION
**Pull request by OniriCorpe**
I just would like the bot to provide the changelog link if available

so I've edited the script to do it if it's set to `latest_github_release`

I haven't tested the script because I have no idea how to configure it properly 🙃

also my code formater touched everything... so tell me if it's a problem

### CODE REVIEW

File: tools/autoupdate_app_sources/autoupdate_app_sources.py [hunk 1]
```diff
@@ -167,9 +162,16 @@ def run(self):
 
             print(f"\n  Checking {source} ...")
 
-            new_version, new_asset_urls = self.get_latest_version_and_asset(
-                strategy, asset, infos, source
-            )
+            if strategy == "latest_github_release" & asset != "tarball":
```
Comment (alexAubin) [inline tools/autoupdate_app_sources/autoupdate_app_sources.py:165]:
```suggestion
            if strategy == "latest_github_release" and asset != "tarball":
```

File: tools/autoupdate_app_sources/autoupdate_app_sources.py [hunk 2]
```diff
@@ -217,8 +229,11 @@ def run(self):
             return bool(todos)
 
         if "main" in todos:
+            if strategy == "latest_github_release" & asset != "tarball":
```
Comment (alexAubin) [inline tools/autoupdate_app_sources/autoupdate_app_sources.py:232]:
```suggestion
            if strategy == "latest_github_release" and asset != "tarball":
```

**Review by alexAubin — COMMENTED** [review]
(super quick review, need to have a look on a specific example, you should be able to run this script on a local folder, though it won't create the corresponding PR, but maybe adding a bunch of debug `print()` would help validate the behavior)

**Comment by OniriCorpe** [comment]
with a `print(message)`

```bash
$ python3 autoupdate_app_sources.py ~/git/ynh/snappymail_ynh/ --commit-and-create-PR

  Checking main ...
Current version in manifest: 2.29.1
Newest  version on upstream: 2.32.0
Update needed for main
Upgrade to v2.32.0
Changelog: https://github.com/the-djmaze/snappymail/releases/tag/v2.32.0
... Branch already exists, skipping

$ sed -i "s/latest_github_release/latest_github_tag/" ~/git/ynh/snappymail_ynh/manifest.toml

$ python3 autoupdate_app_sources.py ~/git/ynh/snappymail_ynh/ --commit-and-create-PR

  Checking main ...
Current version in manifest: 2.29.1
Newest  version on upstream: 2.32.0
Update needed for main
Upgrade to v2.32.0
... Branch already exists, skipping
```

**Comment by alexAubin** [comment]
alrighty let's go then

**Comment by alexAubin** [comment]
woopsies :grimacing:

**Pull request merged**